src/routes/health-check/validation.ts

```diff
@@ -21,10 +21,30 @@ export const EnvSchema = z.object({
 		.string({ message: 'required for emails to be sent' })
 		.min(1, { message: 'required for emails to be sent' }),
 	NODE_ENV: NodeEnvSchema
-});
+}).passthrough();
 
 export type EnvVars = z.infer<typeof EnvSchema>;
 
 export function checkEnvVars(env: Context<EnvironmentBindings>['env']) {
-	return EnvSchema.safeParse(env);
+	const { success, error, data: parsedEnv } = EnvSchema.safeParse(env);
+
+	if (success) {
+		const allEnvKeys = Object.keys(parsedEnv);
+		const usedEnvKeys = Object.keys(EnvSchema.shape);
```

Comment:
```suggestion
		const allEnvKeys = Object.keys(parsedEnv);
		const expectedEnvVars = Object.keys(EnvSchema.shape);
```

I think here we should rename the variable to something else, as it is not listing the unused variables, but the variables that are defined in the schema.

Reply:
sounds good 👍